lib/spack/spack/spec.py

```diff
@@ -1750,7 +1872,7 @@ def do_parse(self):
                     if not specs:
                         self.last_token_error("Dependency has no package")
                     self.expect(ID)
-                    specs[-1]._add_dependency(self.spec())
+                    specs[-1]._add_dependency(self.spec(), (spack.DEP_BUILD, spack.DEP_INSTALL))
```

Comment:
See my comment above on pre-setting the default deptype -- I think this should remain unassigned until `normalize` is called.